packages/nitro-protocol/contracts/ninja-nitro/SingleChannelAdjudicator.sol

```diff
@@ -308,154 +316,117 @@ contract SingleChannelAdjudicator is
-        payouts = new uint256[](indices.length > 0 ? indices.length : allocation.length);
-        totalPayouts = 0;
-        newAllocation = new Outcome.AllocationItem[](allocation.length);
-        safeToDelete = true; // switched to false if there is an item remaining with amount > 0
-        uint256 surplus = initialHoldings; // tracks funds available during calculation
-        uint256 k = 0; // indexes the `indices` array
-
-        // loop over allocations and decrease surplus
-        for (uint256 i = 0; i < allocation.length; i++) {
-            // copy destination part
-            newAllocation[i].destination = allocation[i].destination;
-            // compute new amount part
-            uint256 affordsForDestination = min(allocation[i].amount, surplus);
-            if ((indices.length == 0) || ((k < indices.length) && (indices[k] == i))) {
-                // found a match
-                // reduce the current allocationItem.amount
-                newAllocation[i].amount = allocation[i].amount - affordsForDestination;
-                // increase the relevant payout
-                payouts[k] = affordsForDestination;
-                totalPayouts += affordsForDestination;
-                // move on to the next supplied index
-                ++k;
-            } else {
-                newAllocation[i].amount = allocation[i].amount;
-            }
-            if (newAllocation[i].amount != 0) safeToDelete = false;
-            // decrease surplus by the current amount if possible, else surplus goes to zero
-            surplus -= affordsForDestination;
-        }
+    /**
+     * @notice Pays out assets according to the supplied payouts, only if this channel has been finalized as a guarantor channel, and only if supplied by the __target__ of this channel
+     */
+    function payOutTarget(
+        bytes32 guarantorChannelId,
+        ChannelDataAlt calldata cDL,
+        Outcome.OutcomeItem[] memory payouts
+    ) external {
+        Outcome.OutcomeItem[] memory guarantorOutcome = abi.decode(
+            cDL.outcomeBytes,
+            (Outcome.OutcomeItem[])
+        );
+        Outcome.AssetOutcome memory gAssetOutcome = abi.decode(
+            guarantorOutcome[0].assetOutcomeBytes, // The target channel contract has already checked that all entires have the same target
+            (Outcome.AssetOutcome)
+        );
+        Outcome.Guarantee memory guarantee = abi.decode(
+            gAssetOutcome.allocationOrGuaranteeBytes,
+            (Outcome.Guarantee)
+        );
+        address targetChannelAddress = AdjudicatorFactory(adjudicatorFactoryAddress)
+            .getChannelAddress(guarantee.targetChannelId);
+        require(msg.sender == targetChannelAddress, 'only the target channel is auth');
+        bytes32 outcomeHash = keccak256(cDL.outcomeBytes);
+        _requireMatchingStorage(
+            ChannelData(
```

Comment:
It's hidden away in `IForceMove.sol`, which is not directly imported (only `ForceMove.sol` is imported, which itself imports `ForceMove.sol`). I guess the VSCode extension is not as smart as our compilation toolchain. 

That being said, it may as well live in `ForceMove.sol`, since it is not part of the ForceMove API (just the implementation).

Reply:
f2e2c0d